app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,43 @@
+class CommentsController < ApplicationController
+  before_action :set_problem
+  before_action :set_comment, only: %I[update destroy]
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.problem = @problem
+    @comment.user = current_user
```

Comment:
@comment = Comment.new comment_params.merge(user_id: current_user.id, problem_id: @problem_id)

and remote two next lines